### PR TITLE
IGNITE-8426 LongJVMPauseDetector add logger from context

### PR DESCRIPTION
**Pull request by ezhuravl**

### CODE REVIEW

**Comment by mcherkasov** [comment]
Fix looks good for me.

File: modules/core/src/main/java/org/apache/ignite/internal/LongJVMPauseDetector.java [hunk 1]
```diff
@@ -41,36 +40,43 @@
...
-    private static final IgniteLogger LOG = new JavaLogger();
+    private final IgniteLogger LOG;
 
     /** Worker reference. */
-    private static final AtomicReference<Thread> workerRef = new AtomicReference<>();
+    private final AtomicReference<Thread> workerRef = new AtomicReference<>();
 
     /** Precision. */
-    private static final int PRECISION = getInteger(IGNITE_JVM_PAUSE_DETECTOR_PRECISION, 50);
+    private final int PRECISION = getInteger(IGNITE_JVM_PAUSE_DETECTOR_PRECISION, 50);
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/LongJVMPauseDetector.java]: There is no need to move constants into object.

File: modules/core/src/main/java/org/apache/ignite/internal/LongJVMPauseDetector.java [hunk 2]
```diff
@@ -41,36 +40,43 @@
...
+     */
+    public LongJVMPauseDetector(GridLoggerProxy log) {
+        LOG = log;
+    }
 
     /**
      * Starts worker if not started yet.
      */
-    public static void start() {
+    public void start() {
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/LongJVMPauseDetector.java]: synchronized (LongJVMPauseDetector.class) looks like a bug

File: modules/core/src/main/java/org/apache/ignite/internal/LongJVMPauseDetector.java [hunk 3]
```diff
@@ -41,36 +40,43 @@
...
+    private final long[] longPausesTimestamps = new long[EVT_CNT];
 
     /** Long pauses durations. */
-    private static final long[] longPausesDurations = new long[EVT_CNT];
+    private final long[] longPausesDurations = new long[EVT_CNT];
+
+    /**
+     * @param log Logger.
+     */
+    public LongJVMPauseDetector(GridLoggerProxy log) {
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/LongJVMPauseDetector.java]: GridLoggerProxy can be safely replaced with Ignite Logger.

**Pull request closed without merging**